src/modules/commander/commander.cpp

```diff
@@ -430,6 +433,13 @@ transition_result_t arm_disarm(bool arm, const int mavlink_fd_local, const char
 {
 	transition_result_t arming_res = TRANSITION_NOT_CHANGED;
 
+	// For HIL platforms, require that simulated sensors are connected
+	if (autostart_id >= HIL_ID_MIN && autostart_id <= HIL_ID_MAX
```

Comment:
This check looks like it could be a function ```is_hil_setup()```, as its used two times in this patch: ```autostart_id >= HIL_ID_MIN && autostart_id <= HIL_ID_MAX```.